### PR TITLE
Update classifier.py

### PR DESCRIPTION
**Pull request by PriyankaChaganti**
https://github.com/PriyankaChaganti/traffic_sign_svm/issues/46

### CODE REVIEW

File: robust_svm/classifier.py [hunk 1]
```diff
@@ -135,3 +142,5 @@ def get_svm_label(self, classifier_id, row):
     kernel_type = "linear"
     ml = MultiClassClassifier(training_data,1,{"r0":1,"c":1,kernel_type:"linear"})
     ml.build_classifier(training_data)
+    row = {'class_id':'1','feature_vector':[1, 2, 3]}
+    m1.get_svm_label(row)
```
Comment (sahithi403) [inline robust_svm/classifier.py]: you are supposed to test get_all_classifier_labels

File: robust_svm/classifier.py [hunk 2]
```diff
@@ -127,11 +127,20 @@ def get_svm_label(self, classifier_id, row):
...
 if __name__ == "__main__":
     data_path = training_data_folder
     feature_path = hog3_path
     data_set_list = ['00013', '00015']
     training_data = make_dataset(data_path, feature_path, data_set_list)
     kernel_type = "linear"
     ml = MultiClassClassifier(training_data,1,{"r0":1,"c":1,kernel_type:"linear"})
-    ml.build_classifier(training_data)
+    #ml.build_classifier(training_data)
+    row = {'class_id': '13', 'feature_vector':[0.0178014 , 0.00857185, 0.037246  , ..., 0.0305434 , 0.0267893 ,0.0627695 ]}
```
Comment (sahithi403) [inline robust_svm/classifier.py]: remove '....' from row

File: robust_svm/classifier.py [hunk 3]
```diff
@@ -142,5 +143,5 @@ def get_all_classifier_labels(self,row):
     kernel_type = "linear"
     ml = MultiClassClassifier(training_data,1,{"r0":1,"c":1,kernel_type:"linear"})
     #ml.build_classifier(training_data)
-    row = {'class_id': '13', 'feature_vector':[0.0178014 , 0.00857185, 0.037246  , ..., 0.0305434 , 0.0267893 ,0.0627695 ]}
-    m1.get_all_classifier_labels(row)
+    row = {'class_id': '13', 'feature_vector':[0.0178014]}
```
Comment (sahithi403) [inline robust_svm/classifier.py]: The classifier was trained with feature_vector of length 2916. Therefore, length of w is also 2916

But now you are passing row = {'class_id': '13', 'feature_vector':[0.0178014]} , a feature_vector if length 1

Do row = training_data.data[0] this means that you are accessing first row in training_data (which should be of length 2916)

**Comment by PriyankaChaganti** [comment]
screenshot 6

File: robust_svm/image_processing_utils.py [hunk 4]
```diff
@@ -209,11 +209,11 @@ def get_neighboring_points(x, y, x_lim, y_lim):
...
 if __name__ == "__main__":
-    # # Test the function read_image_annotations
-    # data_class_id='00000'
-    # image_file_name = '00000_00000.ppm'
-    # hog_feature_data= get_hog_features(data_class_id,image_file_name)
-    # print(hog_feature_data)
+    # Test the function read_image_annotations
+    data_class_id='00000'
+    image_file_name = '00000_00000.ppm'
+    hog_feature_data= get_hog_features(data_class_id,image_file_name)
```
Comment (sahithi403) [inline robust_svm/image_processing_utils.py]: There is no 'get_hog_features' function anymore. That function was renamed to something else

Please test your code. before pushing changes

File: robust_svm/classifier.py [hunk 5]
```diff
@@ -127,11 +128,21 @@ def get_svm_label(self, classifier_id, row):
 
         return label
 
+    def get_all_classifier_labels(self,row):
+        results = {}
+        for eachclassifier in self.classifiers:
+            svm_label = self.get_svm_label(eachclassifier,row)
+            results['classifier_id'] = svm_label
+            print(results)
```
Comment (sahithi403) [inline robust_svm/classifier.py]: Do not print the results here. You should print them from name == __main__

Comment (sahithi403) [reply robust_svm/classifier.py]: Post a screenshot of your results

Comment (PriyankaChaganti) [reply robust_svm/classifier.py]: post chesa

Comment (sahithi403) [reply robust_svm/classifier.py]: Where? Will you ever type full sentences?

**Comment by PriyankaChaganti** [comment]
screenshot 8
@sahithi403

File: robust_svm/image_processing_utils.py [hunk 6]
```diff
@@ -209,15 +210,16 @@ def get_neighboring_points(x, y, x_lim, y_lim):
 
 
 if __name__ == "__main__":
-    # # Test the function read_image_annotations
-    # data_class_id='00000'
-    # image_file_name = '00000_00000.ppm'
-    # hog_feature_data= get_hog_features(data_class_id,image_file_name)
-    # print(hog_feature_data)
+    #Test the function read_image_annotations
```
Comment (sahithi403) [inline robust_svm/image_processing_utils.py]: Fix this comment

**Review by sahithi403 — APPROVED** [review]

**Pull request merged**